"staticcheck" tool reveals few problems

```
>staticcheck ./...
literal.go:11:6: don't use Yoda conditions (ST1017)
literal.go:13:13: don't use Yoda conditions (ST1017)
messageformat_test.go:49:13: don't use Yoda conditions (ST1017)
messageformat_test.go:64:15: don't use Yoda conditions (ST1017)
ordinal.go:29:10: assigning the result of this type assertion to a variable (switch v := v.(type)) could eliminate type assertions in switch cases (S1034)
        ordinal.go:36:33: could eliminate this type assertion
ordinal.go:31:11: error strings should not be capitalized (ST1005)
parser.go:41:13: don't use Yoda conditions (ST1017)
parser.go:63:12: don't use Yoda conditions (ST1017)
plural.go:34:6: don't use Yoda conditions (ST1017)
plural.go:35:7: don't use Yoda conditions (ST1017)
plural.go:50:4: this value of err is never used (SA4006)
plural.go:52:7: don't use Yoda conditions (ST1017)
plural.go:59:6: don't use Yoda conditions (ST1017)
plural.go:104:10: assigning the result of this type assertion to a variable (switch v := v.(type)) could eliminate type assertions in switch cases (S1034)
        plural.go:109:29: could eliminate this type assertion
        plural.go:112:36: could eliminate this type assertion
        plural.go:115:16: could eliminate this type assertion
plural.go:106:11: error strings should not be capitalized (ST1005)
plural.go:119:11: assigning the result of this type assertion to a variable (switch v := v.(type)) could eliminate type assertions in switch cases (S1034)
        plural.go:122:22: could eliminate this type assertion
        plural.go:126:36: could eliminate this type assertion
        plural.go:131:22: could eliminate this type assertion
        plural.go:135:36: could eliminate this type assertion
plural.go:121:8: don't use Yoda conditions (ST1017)
plural.go:130:8: don't use Yoda conditions (ST1017)
plural.go:139:8: don't use Yoda conditions (ST1017)
plural.go:182:7: don't use Yoda conditions (ST1017)
select.go:31:13: don't use Yoda conditions (ST1017)
select.go:35:6: don't use Yoda conditions (ST1017)
select.go:81:14: argument char is overwritten before first use (SA4009)
        select.go:82:2: assignment to char
utils.go:11:9: don't use Yoda conditions (ST1017)
utils.go:11:24: don't use Yoda conditions (ST1017)
utils.go:11:40: don't use Yoda conditions (ST1017)
utils.go:11:56: don't use Yoda conditions (ST1017)
utils.go:39:10: assigning the result of this type assertion to a variable (switch v := v.(type)) could eliminate type assertions in switch cases (S1034)
        utils.go:47:30: could eliminate this type assertion
        utils.go:50:11: could eliminate this type assertion
        utils.go:53:29: could eliminate this type assertion
        utils.go:56:35: could eliminate this type assertion
        utils.go:59:35: could eliminate this type assertion
        utils.go:62:35: could eliminate this type assertion
        utils.go:65:29: could eliminate this type assertion
        utils.go:68:37: could eliminate this type assertion
        utils.go:71:37: could eliminate this type assertion
        utils.go:74:37: could eliminate this type assertion
        utils.go:77:37: could eliminate this type assertion
        utils.go:80:30: could eliminate this type assertion
        utils.go:83:39: could eliminate this type assertion
        utils.go:86:31: could eliminate this type assertion
        utils.go:89:29: could eliminate this type assertion
        utils.go:92:29: could eliminate this type assertion
        utils.go:95:31: could eliminate this type assertion
        utils.go:98:11: could eliminate this type assertion
        utils.go:101:11: could eliminate this type assertion
utils_test.go:21:6: func toStringError is unused (U1000)
var.go:26:7: don't use Yoda conditions (ST1017)
```

--- a/plural.go
+++ b/plural.go
@@ -25,7 +25,7 @@ func parsePlural(varname string, ptr_compiler *Parser, char rune, start, end int
 	pos := start + 1
 
 	for pos < end {
-		key, char, i, err := readKey(char, pos, end, ptr_input)
+		key, char, i, err := readKey(pos, end, ptr_input)
 
 		if nil != err {
 			return nil, i, err
@@ -47,7 +47,7 @@ func parsePlural(varname string, ptr_compiler *Parser, char rune, start, end int
 				j++
 			}
 
-			k, c, j, err := readKey(c, j, end, ptr_input)
+			k, c, j, err := readKey(j, end, ptr_input)
 
 			if "" == k {
 				return nil, j, fmt.Errorf("MissingChoiceName")

--- a/select.go
+++ b/select.go
@@ -24,7 +24,7 @@ func parseSelect(varname string, ptr_compiler *Parser, char rune, start, end int
 	pos := start + 1
 
 	for pos < end {
-		key, char, i, err := readKey(char, pos, end, ptr_input)
+		key, char, i, err := readKey(pos, end, ptr_input)
 
 		if nil != err {
 			return nil, i, err
@@ -78,7 +78,7 @@ func formatSelect(expr Expression, ptr_output *bytes.Buffer, data *map[string]in
 	return choice.format(ptr_output, data, ptr_mf, value)
 }
 
-func readKey(char rune, start, end int, ptr_input *[]rune) (string, rune, int, error) {
+func readKey(start, end int, ptr_input *[]rune) (string, rune, int, error) {
 	char, pos := whitespace(start, end, ptr_input)
 	fc_pos, lc_pos := pos, pos
 

